internal/tofu/eval_context_builtin.go

```diff
@@ -11,6 +11,9 @@ import (
 	"log"
 	"sync"
 
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/opentofu/opentofu/internal/lang/marks"
+
 	"github.com/hashicorp/hcl/v2"
 	"github.com/zclconf/go-cty/cty"
```

Comment:
I think these imports need ordering

Reply:
I agree it looks weird.
But when I run `go run golang.org/x/tools/cmd/goimports -w -l internal/tofu/eval_context_builtin.go` it changes nothing (plus, all checks, including the imports ordering, passed for the commit).

Reply:
Hey @Evi1Pumpkin! You need to delete an empty line and rerun sorting. `goimports` treats newlines as a separator for a "group" and thus it sorts imports inside each "group". This is the order I would expect with two "groups":

```
import (
	"context"
	"fmt"
	"log"
	"sync"

	"github.com/hashicorp/hcl/v2"
	"github.com/hashicorp/hcl/v2/hclsyntax"
	"github.com/opentofu/opentofu/internal/addrs"
	"github.com/opentofu/opentofu/internal/checks"
	"github.com/opentofu/opentofu/internal/configs/configschema"
	"github.com/opentofu/opentofu/internal/encryption"
	"github.com/opentofu/opentofu/internal/instances"
	"github.com/opentofu/opentofu/internal/lang"
	"github.com/opentofu/opentofu/internal/lang/marks"
	"github.com/opentofu/opentofu/internal/plans"
	"github.com/opentofu/opentofu/internal/providers"
	"github.com/opentofu/opentofu/internal/provisioners"
	"github.com/opentofu/opentofu/internal/refactoring"
	"github.com/opentofu/opentofu/internal/states"
	"github.com/opentofu/opentofu/internal/tfdiags"
	"github.com/opentofu/opentofu/version"
	"github.com/zclconf/go-cty/cty"
)
```

These "groups" could be configured in IDEs. Here is an example for builtin, local and external "groups":

```
import (
	// builtin
        "context"
	"fmt"
	"log"
	"sync"
	
	// local
	"github.com/opentofu/opentofu/internal/addrs"
	"github.com/opentofu/opentofu/internal/checks"
	"github.com/opentofu/opentofu/internal/configs/configschema"
	"github.com/opentofu/opentofu/internal/encryption"
	"github.com/opentofu/opentofu/internal/instances"
	"github.com/opentofu/opentofu/internal/lang"
	"github.com/opentofu/opentofu/internal/lang/marks"
	"github.com/opentofu/opentofu/internal/plans"
	"github.com/opentofu/opentofu/internal/providers"
	"github.com/opentofu/opentofu/internal/provisioners"
	"github.com/opentofu/opentofu/internal/refactoring"
	"github.com/opentofu/opentofu/internal/states"
	"github.com/opentofu/opentofu/internal/tfdiags"
	"github.com/opentofu/opentofu/version"
	
	// external
	"github.com/hashicorp/hcl/v2"
	"github.com/hashicorp/hcl/v2/hclsyntax"
	"github.com/zclconf/go-cty/cty"
)
```

Reply:
Oh! thanks for the tip :)
fixed it here https://github.com/opentofu/opentofu/pull/1270/commits#:~:text=c2b98f6-,fix%20imports%20order,-%E2%80%A6